client/components/app/App.js

```diff
@@ -80,6 +82,10 @@ class App extends Component {
                   this.handleAuthentication(props);
                   return <Callback {...props} />
                 }} />
+                <Route exact path="/pesquisar" render={(props) => {
+                  this.validateRoute(props.history);
```

Comment:
Evite colocar chamadas de funcões dentro do metodo render,pois cada vez que esse componente for atualizado...a mesma função será chamada.Tem outra solução para esse problema.Criar um componente PrivateRoute que valida dentro dele se o usuário está autenticado.Eu fiz esse componente  esses dias.Se vc quiser te mostro.É simples.